libs/sdk-ui-kit/src/Dialog/ShareDialog/ShareDialogBase/GranteeItem.tsx

```diff
@@ -0,0 +1,112 @@
+// (C) 2021 GoodData Corporation
+import React, { useCallback, useMemo } from "react";
+import { useIntl } from "react-intl";
+import cx from "classnames";
+import {
+    DialogModeType,
+    GranteeItem,
+    IGranteeGroup,
+    IGranteeGroupAll,
+    IGranteeItemProps,
+    IGranteeUser,
+} from "./types";
+import { getGranteeLabel, getGranteeItemTestId } from "./utils";
+import { GranteeGroupIcon, GranteeOwnerRemoveIcon, GranteeRemoveIcon, GranteeUserIcon } from "./GranteeIcons";
+import { Button } from "../../../Button";
+
+interface IGranteeUserItemProps {
+    grantee: IGranteeUser;
+    mode: DialogModeType;
+    onDelete: (grantee: GranteeItem) => void;
+}
+
+interface IGranteeGroupItemProps {
+    grantee: IGranteeGroup | IGranteeGroupAll;
+    mode: DialogModeType;
+    onDelete: (grantee: GranteeItem) => void;
+}
+
+const GranteeUserItem: React.FC<IGranteeUserItemProps> = (props) => {
+    const { grantee, mode, onDelete } = props;
+    const intl = useIntl();
+
+    const granteeLabel = useMemo(() => {
+        const name = getGranteeLabel(grantee, intl);
+        const you = intl.formatMessage({ id: "shareDialog.share.grantee.item.you" });
+        return grantee.isCurrentUser ? `${name} (${you})` : name;
+    }, [grantee, intl]);
+
+    const onClick = useCallback(() => {
+        onDelete(grantee);
+    }, [grantee, onDelete]);
+
+    const itemClassName = cx("gd-share-dialog-grantee-item", getGranteeItemTestId(grantee));
+
+    return (
+        <div className={itemClassName}>
+            {grantee.isOwner ? (
+                <GranteeOwnerRemoveIcon />
+            ) : (
+                <GranteeRemoveIcon mode={mode} onClick={onClick} />
+            )}
+            <div className="gd-grantee-content">
+                <div className="gd-grantee-content-label">{granteeLabel}</div>
+                <div className="gd-grantee-content-label gd-grantee-content-email">
+                    {grantee.granteeEmail}
+                </div>
+            </div>
+            <GranteeUserIcon />
+        </div>
+    );
+};
+
+const GranteeGroupItem: React.FC<IGranteeGroupItemProps> = (props) => {
+    const { grantee, onDelete, mode } = props;
+
+    const intl = useIntl();
+
+    const onClick = useCallback(() => {
+        onDelete(grantee);
+    }, [grantee, onDelete]);
+
+    const groupName = useMemo(() => getGranteeLabel(grantee, intl), [grantee, intl]);
+
+    const numOfUsers = useMemo(() => {
+        if (grantee.granteeCount) {
+            return `${grantee.granteeCount} ${intl.formatMessage({
+                id: "shareDialog.share.grantee.item.users",
```

Comment:
This should be imho handled as parametrized message so translation works correctly: 
`<FormattedMessage id="shareDialog.share.grantee.item.users" params={granteeCount: grantee.granteeCount} />` and change `shareDialog.share.grantee.item.users` to `"{granteeCount} users"` in `en-US.json`.

Reply:
Done